js/fizzbuzz.js

```diff
@@ -0,0 +1,42 @@
+var FizzBuzz = (function () { // Gotta have that IFFE
+
+  var _fizzBuzz = function (x, y) { // Initialize the class/factory/what-have-you
+    this.wordOne = x || "Fizz";
+    this.wordTwo = y || "Buzz";
+  };
+
+  _fizzBuzz.prototype = { // define fizzBuzz methods
+
+    // READING METHOD
+    getValues: function (startVal, endVal) {
+      this.min = startVal;
+      this.max = endVal;
+      this.fizzArray = [];
+      for (var i = startVal; i <= endVal; i++) { // Loop through the values
+        // Check conditions; Push to fizzArray
+        this.fizzArray.push(i % 15 === 0 ? this.wordOne + this.wordTwo : i % 3 === 0 ? this.wordOne : i % 5 === 0 ? this.wordTwo : i);
```

Comment:
I really appreciate the out-of-the-box thinking here, but this is one of those times where ternary operators actually make your code much harder to read.

Better would be to push this in to a function that runs your `if/else` logic:

```javascript
for (var i = startVal; i <= endVal; i++) {
  this.fizzArray.push(getValueFor(i));
}

function getValueFor(idx) {
  if (idx % 15 === 0) {
    return this.wordOne + this.wordTwo
  } else if (idx % 3 === 0) {
    return this.wordOne;
  } else if (idx % 5 === 0) {
    return this.wordTwo;
  } else {
    return idx;
  }
}
```